frontend/src/component/common/DialogFormTemplate/DialogFormTemplate.tsx

```diff
@@ -0,0 +1,112 @@
+import type { FormEventHandler } from 'react';
+import theme from 'themes/theme';
+import {
+    ConfigButtons,
+    ProjectDescriptionContainer,
+    ProjectNameContainer,
+    StyledForm,
+    StyledHeader,
+    StyledInput,
+    TopGrid,
+    LimitContainer,
+    FormActions,
+    styleIcon,
+} from './DialogFormTemplate.styles';
+import { Button } from '@mui/material';
+import { CreateButton } from 'component/common/CreateButton/CreateButton';
+import type { IPermissionButtonProps } from 'component/common/PermissionButton/PermissionButton';
+
+type FormProps = {
+    createButtonProps: IPermissionButtonProps;
+    description: string;
+    errors: { [key: string]: string };
+    handleSubmit: FormEventHandler<HTMLFormElement>;
+    icon: React.ComponentType;
+    Limit?: React.ReactNode;
+    name: string;
+    onClose: () => void;
+    configButtons: React.ReactNode;
+    resource: string;
+    setDescription: (newDescription: string) => void;
+    setName: (newName: string) => void;
+    validateName?: () => void;
+    testIds?: Partial<{
+        nameInput: string;
+        descriptionInput: string;
+        submitButton: string;
+    }>;
+};
+
+export const DialogFormTemplate: React.FC<FormProps> = ({
+    Limit,
```

Comment:
why caps

Reply:
To be consistent with other instances of this and because it's a full component on its own. Again, no strong opinions here, so I'll lowercase it if you want. Or name it `limitComponent` etc.

Reply:
ok caps is fine if component